firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/FirebaseModelDownloaderRegistrar.java

```diff
@@ -45,15 +50,24 @@ public class FirebaseModelDownloaderRegistrar implements ComponentRegistrar {
   @NonNull
   @RequiresApi(api = VERSION_CODES.KITKAT)
   public List<Component<?>> getComponents() {
+    Qualified<Executor> bgExecutor = Qualified.qualified(Background.class, Executor.class);
+    Qualified<ExecutorService> bgExecutorService =
+        Qualified.qualified(Background.class, ExecutorService.class);
+
     return Arrays.asList(
         Component.builder(FirebaseModelDownloader.class)
             .name(LIBRARY_NAME)
             .add(Dependency.required(FirebaseApp.class))
             .add(Dependency.required(FirebaseInstallationsApi.class))
+            .add(Dependency.required(bgExecutor))
+            .add(Dependency.required(bgExecutorService))
             .factory(
                 c ->
                     new FirebaseModelDownloader(
-                        c.get(FirebaseApp.class), c.get(FirebaseInstallationsApi.class)))
+                        c.get(FirebaseApp.class),
+                        c.get(FirebaseInstallationsApi.class),
+                        FirebaseExecutors.newSequentialExecutor(c.get(bgExecutor)),
```

Comment:
cc @rlazo do you know if a sequential executor is required here, or can a multi-threaded executor be used here instead?

Reply:
From reading the code IMO it should be OK to use a multi-threaded executor here.